compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2848,10 +2848,17 @@ class MissingImplicitArgument(
             i"The following implicits in scope can be implicitly converted to ${pt.show}:" +
             ignoredConvertibleImplicits.map { imp => s"\n- ${imp.symbol.showDcl}"}.mkString
           )
+        def importSuggestionAddendum: String =
+          arg.tpe match
+            // If the failure was caused by an underlying NoMatchingImplicits, compute the addendum for its expected type
+            case noMatching: NoMatchingImplicits => // FIXME also handle SynthesisFailure
```

Comment:
As I mentioned in the comment, there are other types of `SearchFailureType` that we may want to handle here.

Reply:
We could add missing cases later if needed.